libcoap-sys/build.rs

```diff
@@ -114,40 +114,55 @@ fn main() {
-                            "-I{} -I{}",
-                            tinydtls_include.to_str().unwrap(),
-                            Path::new(tinydtls_include.to_str().unwrap())
-                                .join("tinydtls")
-                                .to_str()
-                                .unwrap()
-                        ),
-                    );
-                }
+            match dtls_backend {
+                DtlsBackend::TinyDtls => {
+                    // We do not ship tinydtls with our source distribution. Instead, we use tinydtls-sys.
+                    build_config.without("submodule-tinydtls", None);
+                    // If tinydtls-sys is built with the vendored feature, the library is built alongside
+                    // the Rust crate. To use the version built by the tinydtls-sys build script, we use the
+                    // environment variables set by the build script.
+                    if let Some(tinydtls_include) = env::var_os("DEP_TINYDTLS_INCLUDE") {
+                        build_config.env(
+                            "TinyDTLS_CFLAGS",
+                            format!(
+                                "-I{} -I{}",
+                                tinydtls_include.to_str().unwrap(),
+                                Path::new(tinydtls_include.to_str().unwrap())
+                                    .join("tinydtls")
+                                    .to_str()
+                                    .unwrap()
+                            ),
+                        );
+                    };
+                },
+                DtlsBackend::OpenSsl => {
+                    // Set include path according to the path provided by openssl-sys (relevant if
+                    // openssl-sys is vendored)
+                    if let Some(openssl_include) = env::var_os("DEP_OPENSSL_INCLUDE") {
+                        build_config.env("OpenSSL_CFLAGS", format!("-I{}", openssl_include.to_str().unwrap()));
+                    }
+                },
+                DtlsBackend::MbedTls => {
+                    // Set include path according to the path provided by mbedtls-sys (relevant if
+                    // mbedtls-sys is vendored).
+                    // libcoap doesn't support overriding the MbedTLS CFLAGS, but doesnt set those
+                    // either, so we just set CFLAGS and hope they propagate.
+                    if let Some(mbedtls_include) = env::var_os("DEP_MBEDTLS_INCLUDE") {
+                        build_config.env("CFLAGS", format!("-I{}", mbedtls_include.to_str().unwrap()));
+                    }
+                },
+                DtlsBackend::GnuTls => {
+                    // GnuTLS doesn't provide include directory metadata, but doesn't support
+                    // vendoring the library either. Instead, it just uses the GnuTLS found via
+                    // pkg-config, which is the same which is already found by libcoap by default.
```

Comment:
Sentence is a little hard to parse for me, maybe like this?
```suggestion
                    // pkg-config, which is already found by libcoap by default.
```